RRS/accounts/receivers.py

```diff
@@ -0,0 +1,21 @@
+from allauth.account.signals import user_signed_up
+from django.dispatch import receiver
+
+
+@receiver(user_signed_up)
+def user_signed_up_(request, user, sociallogin=None, **kwargs):
+    if sociallogin:
+        if sociallogin.account.provider == 'facebook':
+            user.first_name = sociallogin.account.extra_data['first_name']
+            user.last_name = sociallogin.account.extra_data['last_name']
+ 
+        elif sociallogin.account.provider == 'google':
+            user.first_name = sociallogin.account.extra_data['given_name']
+            user.last_name = sociallogin.account.extra_data['family_name']
+            
+        elif sociallogin.account.provider == 'twitter':
+            name = sociallogin.account.extra_data['name']
+            user.first_name = name.split()[0]
+            user.last_name = name.split()[1]
```

Comment:
I think lines 8-19 can be extracted to another method.

    if sociallogin:
        user.first_name, user.last_name = process_sociallogin(sociallogin.account.provider)